arch-snapshot/arch.md

```diff
@@ -0,0 +1,2921 @@
+    end
+
+    ProviderAgent->>ProviderAgent: Select best Proposals <br/>according to implemented strategy
+    ProviderAgent->>ProviderAgent: Adjust Proposals
+    ProviderAgent->>RequestorAgent: Counter Proposal
+
+    break when the terms of Agreement are satisfactory
+      RequestorAgent->>ProviderAgent: Propose Agreement
+    end
+  end
+  
+  par Proposals from other Nodes in the network
+    GolemNetwork->>ProviderAgent: Receive other Agreement Proposals
+  end
+  ProviderAgent->>ProviderAgent: Select best Agreement Proposal
+  ProviderAgent->>RequestorAgent: Approve Agreement Proposal
+  ProviderAgent->>GolemNetwork: Reject remaining Agreement Proposals
+```
+
+##### Example of negotiation
+
+To better understand the [Negotiation](#process-of-negotiations-and-making-an-agreement) process, let’s consider an example 
+involving the negotiation of a [payment platform](#payment-platform). This will illustrate how agents can use different
+strategies and what negotiation protocols can be built on top of the [property and language](#discovery-and-offersdemand-matching).
+
+When declaring a payment platform in an [Offer](#offer), the Provider Agent lists [wallet](#wallet) addresses for each
+platform it supports. It is the Requestor Agent's responsibility to choose the platform by specifying the appropriate
+[property](#property) in their demand. The Requestor Agent can approach negotiations in two ways:
+
+###### 1. Static Negotiations
+Suppose the Requestor Agent prefers payments on the Polygon network. In this case, they require the Provider Agent to support
+Polygon and will not select a Provider Agent that doesn’t.
+
+Since the Requestor has a specific requirement, multiple negotiation stages aren't necessary. They can simply add
+a [constraint](#constraint) to their [Demand](#demand), instructing the matching algorithm to filter out Providers that
+don’t meet this requirement. In their Demand, they set the chosen platform as a fixed value.
+
+###### 2. Dynamic Negotiations
+Now imagine a Requestor Agent that can pay on multiple platforms but prioritizes them based on transaction fees. In this
+scenario, the Requestor Agent has a larger pool of potential Providers since they don’t restrict the platform by adding
+a constraint to their demand.
+
+Instead, the Requestor Agent collects Proposals from the market and evaluates them based on estimated costs. In later stages
+of Proposal exchange, they choose the platform by setting the relevant [property](#property) according to the Providers' scores,
+which are based on potential [transaction](#transaction-on-blockchain) costs.
+
+#### 4. Allocate the promised Resources in accordance with the Agreement
+
+Once the [Agreement](#agreement) is signed, the Provider Agent is expected to reserve the promised [resources](#resource) for the
+Requestor’s use. During this time, the Provider Agent cannot sell these resources to anyone else and must be prepared
```

Comment:
Reserve for how long? Indefinitely? What prevents an evil requestor from blocking provider resources without starting an Activity (and hence without payment)?